float64 left_angle
float64 right_angle
